### PR TITLE
Add tree operations test vector

### PR DESCRIPTION
**Pull request by mulmarta**
As discussed, simple tests for applying operations to the tree. The trees were generated using cipher suite `1`; let me know if this info is useful and I should add it to description.

### CODE REVIEW

**Review by bifurcation — APPROVED** [review]
This structure looks good to me.  I will try to verify later today.

I'm a little split on whether to have a `cipher_suite` field.  On the one hand, it shouldn't matter to the tests, and we don't need to generate multiple for different ciphersuites.  On the other hand, it could be handy if there are stacks a tree representation wants to know what ciphersuite to use (I don't think MLSpp is among these).  Net/net, I would probably include it.

**Comment by TWal** [comment]
Just a quick note: the ratchet tree serialization is malleable, because the protocol document says implementations SHOULD (not MUST) omit blank nodes at the right end of the tree.

**Comment by bifurcation** [comment]
I would be open to a spec PR that would change that SHOULD to MUST.

**Comment by TWal** [comment]
Shouldn't this test vector also test applying UpdatePath?
There are some non-trivial things happening there, with the parent-hash computation.
Or do you consider it's something that belong to the TreeKEM test vector?

**Comment by bifurcation** [comment]
I think that's covered in the TreeKEM test vector.

File: test-vectors.md [hunk 1]
```diff
@@ -457,6 +457,29 @@ Verification:
...
+
+  // Computed values
+  "tree_after": /* hex-encoded TLS-serialized ratchet tree */,
+}
+```
+
+The type of `proposal` is either `add`, `remove` or `update`. 
+
+Verification:
+* Compute `candidate_tree_after` byt applying `proposal` sent by the member
```
Comment (bifurcation) [inline test-vectors.md:478]:
```suggestion
* Compute `candidate_tree_after` by applying `proposal` sent by the member
```

**Pull request merged**